### PR TITLE
WRKLDS-950: use must-gather since and since time env vars on inspect cmds

### PR DESCRIPTION
**Pull request by flavianmissi**

### CODE REVIEW

File: collection-scripts/gather
Comment (ardaguclu) [inline collection-scripts/gather:54]: Should we handle the cases where `MUST_GATHER_SINCE` or `MUST_GATHER_SINCE_TIME` are unset such as?;

```sh
oc adm inspect --since= --since-time= --dest-dir must-gather
```

Comment (flavianmissi) [reply collection-scripts/gather:54]: When `--since` and `--since-time` are unset, `oc adm must-gather` will still set both envs with Go's empty value for each type, so if the client doesn't specify either flag they will be `0s` and an empty string respectively. My manual testing show that this works.
This approach isn't the most clear though... WDYT?

Comment (ardaguclu) [reply collection-scripts/gather:54]: Yeah, I agree that this isn't clear. Besides if there is any change in defaults, these commands would fail. I think, we can only pass these flags when they are set.

Comment (flavianmissi) [reply collection-scripts/gather]: Latest push addresses your comment @ardaguclu, PTAL.

File: collection-scripts/common.sh
Comment (ardaguclu) [inline collection-scripts/common.sh:23]: Since must-gather command in oc is the caller and validation is done here, we don't have to validate that MUST_GATHER_SINCE and MUST_GATHER_SINCE_TIME are both set.

**Review by ardaguclu — COMMENTED** [review]
I have a question, other than that it looks good to me. Thanks.

File: collection-scripts/common.sh
Comment (ardaguclu) [reply collection-scripts/common.sh:23]: Just a note for the future reviewers :)

Comment (flavianmissi) [reply collection-scripts/common.sh:23]: Ah yes, that is a good note to make. Added a comment to the code (will push soon)

File: collection-scripts/gather_service_logs_util
Comment (flavianmissi) [reply collection-scripts/gather_service_logs_util:19]: huh I must have fat fingered this, good catch - rolling it back to $1.

**Comment by ardaguclu** [comment]
LGTM. I'm deferring to @ingvagabund for another set of eyes.

**Comment by flavianmissi** [comment]
something is still off, holding until I figure it out
/hold

**Comment by flavianmissi** [comment]
Checked that `--since-time` works for both node logs (under `host_service_logs`) and pod logs (under `namespaces/.../pod/.../logs/current.log`.
`--since` is a little more complicated - when a duration is given from via the cli as `2h`, after calling `duration.String()` it is formatted as `2h0m0s` and `oc adm node-logs` does takes a duration matching the `(+|-)[0-9]+(s|m|h|d)` regex. I'm going to have to truncate the duration string to keep its left most part, so in the case of `2h0m0s` only `2h` would be given as an argument to `oc adm node-logs`.

**Comment by flavianmissi** [comment]
fixed formatted of duration given to `oc adm node-logs`.
also using lowercase now for the non-env-vars, which seem to be the best practice.
will be pushing the updates in a moment.

**Comment by flavianmissi** [comment]
Both flags now work as they should! 
/hold cancel

**Comment by flavianmissi** [comment]
@ardaguclu this one is ready to go in!

**Comment by ardaguclu** [comment]
Thanks @flavianmissi 
/lgtm

File: collection-scripts/gather_audit_logs
Comment (flavianmissi) [inline collection-scripts/gather_audit_logs:51]: this doesn't actually work - `xargs` does not have access to env vars, so `node_log_collection_args` will always be empty here. right now it doesn't make a difference because the `--since` flag only affects jounal logs for `oc adm node-logs`. 
I'm fixing it anyway so we're future proof.

Comment (flavianmissi) [reply collection-scripts/gather_audit_logs]: last push fixes this - @ardaguclu can you take another look? This is the last one, I promise 😅

Comment (ardaguclu) [reply collection-scripts/gather_audit_logs]: No worries :). Tagged.

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**
